clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -1489,119 +1491,112 @@ def create_file(
                 compress=True if zip_image == "y" else False,
                 bids_sidecar=True if generate_json == "y" else False,
             )
-    else:
-        file_without_extension = output_path / output_filename
-        output_image = file_without_extension.with_suffix(".nii.gz")
 
-        if image.Is_Dicom:
-            success = run_dcm2niix(
-                input_dir=image_path,
-                output_dir=output_path,
-                output_fmt=output_filename,
-                compress=True if zip_image == "y" else False,
-                bids_sidecar=True if generate_json == "y" else False,
+    if image.Is_Dicom:
```

Comment:
If the modality is "fmap", this will also get executed. I don't think anything will happen since `run_dcm2niix` was already run but it might be better to reorganize things a bit. Maybe something along those lines:

```python
if image.Is_Dicom:
    if modality == "fmap":
        success = all([run_dcm2niix(...) for fmap_image_path in Path(image_path).parent.iterdir()])
    else:
        success = run_dcm2niix(...)
    if not success:
        cprint(....)
for trigger_time in output_path.glob(f"{output_filename}_t[0-9]*"):
    ....
```

WDYT ?

Reply:
> If the modality is "fmap", this will also get executed. I don't think anything will happen since `run_dcm2niix` was already run but it might be better to reorganize things a bit. Maybe something along those lines:
> 
> ```python
> if image.Is_Dicom:
>     if modality == "fmap":
>         success = all([run_dcm2niix(...) for fmap_image_path in Path(image_path).parent.iterdir()])
>     else:
>         success = run_dcm2niix(...)
>     if not success:
>         cprint(....)
> for trigger_time in output_path.glob(f"{output_filename}_t[0-9]*"):
>     ....
> ```
> 
> WDYT ?

You're right, I went a little fast on this